include/seqan3/range/decorator/gap_decorator.hpp

```diff
@@ -597,6 +777,33 @@ class gap_decorator
     {
         return const_iterator{*this, size()};
     }
+
+    /*!\brief Returns a random access iterator pointing behind the last element of the decorator.
+     * \returns Iterator pointing behind the last element.
+     *
+     * \attention The returned iterator models the std::ranges::RandomAccessIterator but
+     *            **the operations on it do not guarantee constant time**.
+     *
+     * \attention This element acts as a placeholder; attempting to dereference it results in undefined behaviour.
+     *
+     * ### Complexity
+     *
+     * Constant.
+     *
+     * ### Exceptions
+     *
+     * No-throw guarantee.
+     */
+    iterator_ra end_ra() const noexcept
+    {
+        return gap_decorator_iterator_ra{*this, size()};
+    }
+
```

Comment:
missing `end_ra()` non const version and the const version should return const_iterator. For the docu

Reply:
But now end_ra should return a const_iterator because it is const. Actually why is there a const_iteator and a iterator type if they are the same. I suggest to just remove the const_iterator type and make clear in the documentation that there is no non-const version of it.

Reply:
hm but isn't the member type `const_iterator` part of the (although not enforced) interface of a container/decorator? I can remove it but I am not sure about the usability.